### PR TITLE
Add experimental dual stack profile

### PR DESCRIPTION
**Pull request by jacob-delgado**
**Please provide a description of this PR:**

Add a profile for users to install Istio with dual-stack experimental support. This is based on the default profile.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [X] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: manifests/profiles/dualstack.yaml [hunk 1]
```diff
@@ -0,0 +1,179 @@
+apiVersion: install.istio.io/v1alpha1
```
Comment (howardjohn) [inline manifests/profiles/dualstack.yaml:1]: can we keep the profile to just the minimum set of changes needed to get it working? 99% of this is unrelated

Comment (jacob-delgado) [reply manifests/profiles/dualstack.yaml:1]: Done

File: releasenotes/notes/dualstack-profile.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+issue: []
+
+releaseNotes:
+- |
+  **Added** profile to install Istio in dual-stack experimental mode.
```
Comment (ericvn) [inline releasenotes/notes/dualstack-profile.yaml:8]: Do we want to call out the name as well?
```suggestion
  **Added** dual stack profile to install Istio in dual-stack experimental mode.
```

File: manifests/profiles/dualstack.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+kind: IstioOperator
```
Comment (ymesika) [inline manifests/profiles/dualstack.yaml:1]: Don't you miss the `apiVersion: install.istio.io/v1alpha1` top declaration here?

**Review by howardjohn — COMMENTED** [review]
Do we really need this? I don't want too many profiles for every little feature. Maybe we can just add it to `preview.yaml`, which is "new things that are not yet stable"?

**Comment by jacob-delgado** [comment]
> Do we really need this? I don't want too many profiles for every little feature. Maybe we can just add it to `preview.yaml`, which is "new things that are not yet stable"?

It can go either way. Ideally once we get enough test coverage there won't be a feature gate (i.e., auto-detection of dual-stack) behind this. We can also expose it as a `meshConfig` value, but we've proposed not to do this too. People that need dual-stack Istio need to find out a way to expose this. Why don't I just create a page on istio.io for this instead? Right now we only have the blog.
